lib/active_shipping/carriers/canada_post_pws.rb

```diff
@@ -577,19 +621,27 @@ def tracking_url(pin)
 
     def create_shipment_url(options)
       raise MissingCustomerNumberError unless customer_number = options[:customer_number]
-      if @platform_id.present?
-        endpoint + "rs/#{customer_number}-#{@platform_id}/ncshipment"
+      if contract?
+        endpoint + "rs/#{customer_number}/#{customer_number}/shipment"
```

Comment:
The second `customer_number` is actually the `mobo` (mail on behalf of). If specified it should be used or repeat the `customer_number` 

From the docs
>the mailed on behalf of customer number or repeat your customer number